Makefile

```diff
@@ -1,7 +1,24 @@
-ST2_VERSION ?= 3.4dev
+ST2_VERSION ?= 3.2.0
 DOCKER_TAG ?= ${ST2_VERSION}
+RELEASE_TAG_REGEX := [^dev]$$
 SHELL := /bin/bash
 
+# supported values of the TAG_UPDATE_FLAG
+# 0 = no additional tags to be set
+# 1 = add the major.minor tag
+# 2 = add the tags major and major.minor
+# 3 = add the tags major, major.minor and latest
+TAG_UPDATE_FLAG := $(shell ./determine_needed_tags.sh st2 ${ST2_VERSION})
```

Comment:
I think it would be best to just scan the exit code from the script and fail fast if it's non-zero.

Reply:
Now all failures in the script should be handled and return a message with the "Error:"-Prefix and the makefile checks the output for the Error-Prefix and fails before building any images. 

Another idea was to introduce another return value like -1 in case of an error but then I wouldn't know how to show the error message in the CI logs. If you have an idea to show the output from the script in the CI log please let me know. 

I just pushed a Makefile with 3.3.0 and will change it back to 3.4dev before merging this branch. That's just because the determine tags script is skipped if we build a *dev version and not a release.

Reply:
Depending on "Error" message as a failure flag adds some knowledge overhead in requirements that could be broken in the future.
What if 3rd Maintainer comes, edit the `determine_needed_tags.sh` script and doesn't add "Error" message for the failure case.

We can at least reinforce it with `determine_needed_tags.sh || echo Error`  to fit your solution.